images/router/haproxy/conf/haproxy-config.template

```diff
@@ -228,6 +231,27 @@ frontend fe_sni
   # before matching, or any requests containing uppercase characters will never match.
   http-request set-header Host %[req.hdr(Host),lower]
 
+  {{- if ne (env "ROUTER_MUTUAL_TLS_AUTH" "none") "none" }}
+    {{- with (env "ROUTER_MUTUAL_TLS_AUTH_CN") }}
+  # If a mutual TLS auth CN is set, we deny requests if the common name doesn't
+  # match. A custom template can change this behavior (e.g. set custom headers).
```

Comment:
Please specify exactly what name the cert needs to match

Reply:
Common Name ... will update this to make this clearer.

Reply:
I am not asking what CN is, I am asking what it should match, it is not clear to me what you have to set CN to.

Reply:
This sounds wrong, we should probably have a ROUTER_MUTUAL_TLS_AUTH_DN variable and match the exact subject as a default.
In what case matching a substring of a random CN in the subject is useful ?

Reply:
That's what the customer was using/asking - ref: https://bugzilla.redhat.com/show_bug.cgi?id=1477349

In any case, this is moot because I just changed this to be an auth filter (see latest commit) to filter based on the subject (substring) so you can run a router ala: 
```
oc adm router --latest-images=true  --mutual-tls-auth=optional  \
              --mutual-tls-auth-ca=_ramr/nodejs-header-echo/config/CA/cacert.pem  \
              --mutual-tls-auth-filter="CN=header.test"
```

As re: an exact match, see the other comment re: filtering on organizations or OUs.